lib/utilities.js

```diff
@@ -13,4 +13,33 @@ async function getConnectedDevices (socket) {
   }
 }
 
-export { getConnectedDevices };
+async function getOSVersion (udid, socket) {
+  const usbmux = new Usbmux(socket);
+  try {
+    // lockdown doesn't need to be closed since it uses the same socket usbmux uses
+    const lockdown = await usbmux.connectLockdown(udid);
+    return await lockdown.getValue({ Key: 'ProductVersion' });
+  } finally {
+    usbmux.close();
+  }
+}
+
+async function startLockdownSession (udid, socket) {
+  const usbmux = new Usbmux(socket);
+  try {
+    const pairRecord = await usbmux.readPairRecord(udid);
+    if (!pairRecord) {
+      throw new Error(`Couldn't find a pair record for device ${udid}. Please firt pair with the device`);
+    }
+    // lockdown doesn't need to be closed since it uses the same socket usbmux uses
+    const lockdown = await usbmux.connectLockdown(udid);
+    await lockdown.startSession(pairRecord.HostID, pairRecord.SystemBUID);
+    lockdown.enableSessionSSL(pairRecord.HostPrivateKey, pairRecord.HostCertificate);
+    return lockdown;
+  } catch (e) {
+    usbmux.close();
```

Comment:
This should be in a `finally` block, or it won't be closed unless there's an error.

Reply:
That is the point. It is for passing the lockdown object to the high level caller. We can decide if we want to design the api like this but the PR is doing the expected scenario

Reply:
But if we're not also passing `usbmux` to the caller, how could it ever be closed? If this kind of thing should be passed around, probably the usbmux connection should be a field on the lockdown object, so it can be managed by whomever's managing lockdown.

Reply:
@jlipps The only thing that needs to be closed here is the socketClient and it is the socketClient when you upgrade the connection from usbmux to lockdown. It kinda works like socket to ssl socket upgrade. It doesn't matter whichever you close since it is the tcp connection you need to close. The same thing applies here as well

I don't want to put `usbmux` as a field to `lockdown` because it has no functionality there since the connection is already upgraded and can't be downgraded to a `usbmux` protocol again